cache/contenthash/filehash.go

```diff
@@ -33,6 +33,9 @@ func NewFileHash(path string, fi os.FileInfo) (hash.Hash, error) {
 		stat.Mode = stat.Mode | 0777
 	}
 
+	// Clear the socket bit since archive/tar.FileInfoHeader does not handle it
+	stat.Mode &^= uint32(os.ModeSocket)
```

Comment:
would this make more sense inside `NewFromStat` ?

Reply:
If you think that's a better place we can definitely move it there.

in `NewFromStat(stat *fstypes.Stat)` stat is passed as a pointer, would it be fine to modify the contents, or would it be better to make a copy of it so it's essentially passed in as a const?

Reply:
I'm fine with the current as it is a trivial case, but if you want to be more correct you could check it the socket bit exists and only do copy then.